src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/CompletionListCache.cs

```diff
@@ -32,41 +34,58 @@ public int Set(VSInternalCompletionList completionList, object? context)
 
         lock (_accessLock)
         {
-            // If cache exceeds maximum size, remove the oldest list in the cache
-            if (_resultIdToCacheEntry.Count >= MaxCacheSize)
+            var index = _nextIndex++;
+            var id = _nextId++;
+
+            _items[index] = new Slot(id, completionList, context);
+
+            if (_nextIndex == MaxCacheSize)
```

Comment:
This is a super-nit, but it did my head in for a second that this check was after the array access. I had to concentrate 😛